$NetBSD: patch-third__party_libwebrtc_modules_desktop__capture_desktop__capture__gn_moz.build,v 1.3 2025/09/03 14:17:01 ryoon Exp $

* Disable Wayland desktop capture for non-Linux platforms.
  Fix segfault under X11.

--- third_party/libwebrtc/modules/desktop_capture/desktop_capture_gn/moz.build.orig	2025-08-23 02:57:35.395483264 +0000
+++ third_party/libwebrtc/modules/desktop_capture/desktop_capture_gn/moz.build
@@ -418,7 +418,7 @@ if CONFIG["MOZ_DEBUG"] == "1" and CONFIG
 
     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
 
-if CONFIG["TARGET_CPU"] == "aarch64" and (CONFIG["OS_TARGET"] == "Linux" or CONFIG["OS_TARGET"] == "NetBSD" or CONFIG["OS_TARGET"] == "OpenBSD" or CONFIG["OS_TARGET"] == "FreeBSD"):
+if CONFIG["TARGET_CPU"] == "aarch64" and CONFIG["OS_TARGET"] == "Linux":
 
     DEFINES["CR_SYSROOT_KEY"] = "20250129T203412Z-1"
     DEFINES["WEBRTC_USE_PIPEWIRE"] = True
@@ -470,7 +470,7 @@ if CONFIG["OS_TARGET"] == "Linux" and CO
         "/third_party/libwebrtc/modules/desktop_capture/linux/x11/x_window_property.cc"
     ]
 
-if CONFIG["TARGET_CPU"] == "x86" and (CONFIG["OS_TARGET"] == "Linux" or CONFIG["OS_TARGET"] == "NetBSD" or CONFIG["OS_TARGET"] == "OpenBSD" or CONFIG["OS_TARGET"] == "FreeBSD"):
+if CONFIG["TARGET_CPU"] == "x86" and CONFIG["OS_TARGET"] == "Linux":
 
     CXXFLAGS += [
         "-msse2"
@@ -499,7 +499,7 @@ if CONFIG["TARGET_CPU"] == "x86" and (CO
         "/third_party/libwebrtc/modules/desktop_capture/linux/wayland/shared_screencast_stream.cc"
     ]
 
-if CONFIG["TARGET_CPU"] == "x86_64" and (CONFIG["OS_TARGET"] == "Linux" or CONFIG["OS_TARGET"] == "NetBSD" or CONFIG["OS_TARGET"] == "OpenBSD" or CONFIG["OS_TARGET"] == "FreeBSD"):
+if CONFIG["TARGET_CPU"] == "x86_64" and CONFIG["OS_TARGET"] == "Linux":
 
     DEFINES["CR_SYSROOT_KEY"] = "20250129T203412Z-1"
     DEFINES["WEBRTC_USE_PIPEWIRE"] = True
@@ -663,6 +663,10 @@ if CONFIG["TARGET_CPU"] == "x86_64" and 
 
     DEFINES["WEBRTC_USE_X11"] = True
 
+    CXXFLAGS += [
+        "-msse2"
+    ]
+
     OS_LIBS += [
         "X11",
         "Xcomposite",
